### ISSUE TITLE
Bug: (fill in) structuredClone is not defined

### ISSUE THREAD

**Issue by varxzy** [eslint/eslint#18851]
### Environment

Node version: 20.17.0
npm version: 10.8.2
Local ESLint version: 9.9.1
Global ESLint version: null
Operating System: macOS Catalina 10.15.7

### What parser are you using?

Default (Espree)

### What did you do?

<summary>eslint.config.js configuration</summary>
Default configuration used.

```
import globals from "globals";
import pluginJs from "@eslint/js";
import tseslint from "typescript-eslint";
import pluginVue from "eslint-plugin-vue";


export default [
  {files: ["**/*.{js,mjs,cjs,ts,vue}"]},
  {languageOptions: { globals: {...globals.browser, ...globals.node} }},
  pluginJs.configs.recommended,
  ...tseslint.configs.recommended,
  ...pluginVue.configs["flat/essential"],
  {files: ["**/*.vue"], languageOptions: {parserOptions: {parser: tseslint.parser}}},
];

```

<summary>Command line input</summary>

```
pnpm dlx eslint
```

### What did you expect to happen?

Correctly execute the eslint program.

### What actually happened?

<summary>Command line output</summary>

```
Packages: +89
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
Progress: resolved 89, reused 89, downloaded 0, added 89, done

Oops! Something went wrong! :(

ESLint: 9.9.1

ConfigError: Config (unnamed): Key "rules": Key "constructor-super": structuredClone is not defined
    at rethrowConfigError (/Volumes/Dev/.pnpm-store/v3/tmp/dlx-1505/node_modules/.pnpm/@eslint+config-array@0.18.0/node_modules/@eslint/config-array/dist/cjs/index.cjs:303:8)
    at /Volumes/Dev/.pnpm-store/v3/tmp/dlx-1505/node_modules/.pnpm/@eslint+config-array@0.18.0/node_modules/@eslint/config-array/dist/cjs/index.cjs:1098:5
    at Array.reduce (<anonymous>)
    at FlatConfigArray.getConfigWithStatus (/Volumes/Dev/.pnpm-store/v3/tmp/dlx-1505/node_modules/.pnpm/@eslint+config-array@0.18.0/node_modules/@eslint/config-array/dist/cjs/index.cjs:1091:43)
    at FlatConfigArray.getConfig (/Volumes/Dev/.pnpm-store/v3/tmp/dlx-1505/node_modules/.pnpm/@eslint+config-array@0.18.0/node_modules/@eslint/config-array/dist/cjs/index.cjs:1120:15)
    at /Volumes/Dev/.pnpm-store/v3/tmp/dlx-1505/node_modules/.pnpm/eslint@9.9.1/node_modules/eslint/lib/eslint/eslint-helpers.js:346:56
    at Array.reduce (<anonymous>)
    at /Volumes/Dev/.pnpm-store/v3/tmp/dlx-1505/node_modules/.pnpm/eslint@9.9.1/node_modules/eslint/lib/eslint/eslint-helpers.js:333:36
    at /Volumes/Dev/.pnpm-store/v3/tmp/dlx-1505/node_modules/.pnpm/eslint@9.9.1/node_modules/eslint/lib/eslint/eslint-helpers.js:296:32
    at Object.isAppliedFilter (/Volumes/Dev/.pnpm-store/v3/tmp/dlx-1505/node_modules/.pnpm/@nodelib+fs.walk@1.2.8/node_modules/@nodelib/fs.walk/out/readers/common.js:12:31)
```

### Link to Minimal Reproducible Example

null

### Participation

- [x] I am willing to submit a pull request for this issue.

**Comment by varxzy** [comment]
The `npx eslint` command can be executed correctly.

**Comment by Tanujkanti4441** [comment]
Hi @varxzy, thanks for the issue, but i am not able to reproduce it with the info you have given in issue template.

here is i tried to reproduce it => [stackblitz](https://stackblitz.com/edit/stackblitz-starters-nkgpvo?description=Starter%20project%20for%20Node.js,%20a%20JavaScript%20runtime%20built%20on%20Chrome%27s%20V8%20JavaScript%20engine&file=index.js&title=node.new%20Starter)

please share if  i missed something or any other information regarding this.

**Comment by prajilshrestha** [comment]
@Tanujkanti4441 I am having the same issue, mine was created after I ran the migration from .eslintrc.cjs react/typescript. My research says its due to the airbnb package not supporting the latest breaking changes.

**Comment by Tanujkanti4441** [comment]
> [@Tanujkanti4441](https://github.com/Tanujkanti4441) I am having the same issue, mine was created after I ran the migration from .eslintrc.cjs react/typescript. My research says its due to the airbnb package not supporting the latest breaking changes.
> 
> `ConfigError: Config (unnamed): Key "rules": Key "accessor-pairs": structuredClone is not defined`

@prajilshrestha, could you share a reproducible example through [stackblitz](https://stackblitz.com/) or some other way?

**Comment by prajilshrestha** [comment]
@Tanujkanti4441 I have created an issue report, this is the same config that gives me the structuredClone warning on my IDE (Webstorm).
#18852

**Comment by aladdin-add** [comment]
[`structuredClone` should already be supported in node.js v17.](https://2ality.com/2022/01/structured-clone.html#on-which-javascript-platforms-is-structuredclone()-available%3F) So it should be that you were using an older version of node.js, not eslint itself.

**Comment by varxzy** [comment]
> [`structuredClone` should already be supported in node.js v17.](https://2ality.com/2022/01/structured-clone.html#on-which-javascript-platforms-is-structuredclone()-available%3F) So it should be that you were using an older version of node.js, not an eslint issue.

I use volta to manage nodejs versions, and the current nodejs version is v20.17.0. I tried to execute `console.log(structuredClone({}))` in the command line, and it worked correctly.

In addition, `npx eslint` can be executed correctly, but `pnpm dlx eslint` has an error.

**Comment by varxzy** [comment]
> Hi [@varxzy](https://github.com/varxzy), thanks for the issue, but i am not able to reproduce it with the info you have given in issue template.
> 
> here is i tried to reproduce it => [stackblitz](https://stackblitz.com/edit/stackblitz-starters-nkgpvo?description=Starter%20project%20for%20Node.js,%20a%20JavaScript%20runtime%20built%20on%20Chrome%27s%20V8%20JavaScript%20engine&file=index.js&title=node.new%20Starter)
> 
> please share if i missed something or any other information regarding this.

@Tanujkanti4441 I tried to create a vite-vue project on the stackblitz platform, but the problem did not reproduce. The nodejs version specified by the stackblitz platform is v18.20.3, and I cannot change the nodejs version.

So I am not sure whether it is a problem with eslint or a problem with my workbench environment.

Currently, I use the npx eslint command, which works normally for the time being.

**Comment by varxzy** [comment]
> Hi [@varxzy](https://github.com/varxzy), thanks for the issue, but i am not able to reproduce it with the info you have given in issue template.
> 
> here is i tried to reproduce it => [stackblitz](https://stackblitz.com/edit/stackblitz-starters-nkgpvo?description=Starter%20project%20for%20Node.js,%20a%20JavaScript%20runtime%20built%20on%20Chrome%27s%20V8%20JavaScript%20engine&file=index.js&title=node.new%20Starter)
> 
> please share if i missed something or any other information regarding this.

@Tanujkanti4441 Sorry, I know what caused this problem. It's not an eslint problem, but a problem with volta's support for pnpm.

I use Volta to manage nodejs versions. The nodejs version set for my current project is v20.17.0, so the `npx eslint` command can be executed correctly. When I installed pnpm, the nodejs version at that time was v16.20.3, which resulted in the nodejs version that pnpm depends on still being v16.20.3 when executing pnpm-related commands in node@20.17.0. I don't know why this happened for the time being, but it's obviously a problem with Volta's support for pnpm, not an eslint problem.

I reinstalled pnpm using Volta in the nodejs@20.17.0 environment, and the problem of `structuredClone is not defined` did not occur again.

Thanks.

**Comment by aladdin-add** [comment]
glad you got it :). closing as no action needed here.

**Comment by maingocthanhtan96** [comment]
The same issue.  I fixed

```js
if (typeof global.structuredClone === 'undefined') {
  global.structuredClone = obj => JSON.parse(JSON.stringify(obj));
}
```

**Comment by ShuBuShao** [comment]
原因可能是node版本过低导致的，可尝试使用node20

**Comment by yaaminu** [comment]
If you're facing this problem despite having tried everything here, the fundamental cause of this error is having an older version nodjes than eslint expects. 

To fix this, you have to first install a newer version of nodejs that supports structed clone. If you're using eslint in an IDE, configure the IDE to use the newer version of node you installed. If the issue still persists, restart the IDE and that should resolve the issue.

**Comment by frank-topel-dbi** [comment]
Also having this issue:

```
✖ eslint:

Oops! Something went wrong! :(

ESLint: 9.17.0

ConfigError: Config (unnamed): Key "rules": Key "constructor-super": structuredClone is not defined
```
